tesseract_python/tesseract_python/swig/tesseract_environment/core/environment.i

```diff
@@ -150,5 +145,28 @@ public:
   registerContinuousContactManager(const std::string& name,
                                    tesseract_collision::ContinuousContactManagerFactory::CreateMethod create_function);
 
+  %extend
```

Comment:
I am not very familiar with SWIG and writting python wrappers but does python not support the ability to move objects?

Reply:
If not, is there a reason not to use `const Joint& joint` for these function, it would be more flexible to the python developer so they do not have to create a shared pointer?

Reply:
`Joint` is always a shared pointer in Python so it shouldn't make much difference. The only wrinkle is that shared pointers can be null so that needs to be checked.

Reply:
Indeed @Levi-Armstrong : in Python everything is a pointer anyways, and there is no such thing as "moving"... So I settled for this as a compromise.

Reply:
Ah, Can you tell I don't use python much :). Looks good.